portfolio/src/main/java/com/google/sps/data/ServerResponse.java

```diff
@@ -0,0 +1,14 @@
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ServerResponse {
+  public static ServerResponse create(@Nullable String email, String url) {
+    return new AutoValue_ServerResponse(email, url);
+  }
+  
+  @Nullable abstract String email();
+  abstract String url();
```

Comment:
Since "url" is such a generic name, add a comment to explain what this url is (i.e. logout url if email is not null, login url if email is null)